pulp_service/pulp_service/app/viewsets.py

```diff
@@ -164,3 +165,14 @@ def post(self, request):
         repo_version_pk = serialized_data.data["repo_version"]
         task = dispatch(check_content, kwargs={"repo_version_pk": repo_version_pk})
         return OperationPostponedResponse(task, request)
+
+
+class NPMDependencyVulnerabilityReport(ViewSet):
+
+    def post(self, request):
+        """Dispatch a task to scan the npm dependencies' vulnerabilities"""
+        serialized_data = NPMPackageLockSerializer(data=request.data)
+        serialized_data.is_valid(raise_exception=True)
+        file_content = serialized_data.verify_file()
+        task = dispatch(check_content, kwargs={"npm_package": file_content})
```

Comment:
We try to avoid passing "large" data via arguments to tasks. This is actually where you want to use `PulpTemporaryFile` to save it and then load it in the tasking code.